src/main/java/club/issizler/okyanus/api/cmdnew/CommandEnvelope.java

```diff
@@ -0,0 +1,104 @@
+package club.issizler.okyanus.api.cmdnew;
+
+import club.issizler.okyanus.api.cmdnew.mck.MckCommandRunnable;
+import org.cactoos.list.ListOf;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+abstract class CommandEnvelope implements ICommand {
+
+    private final String id;
+    private final List<CommandOption> commandOptions;
+    private final List<Requirement> requirements = new ArrayList<>();
+
+    CommandEnvelope(String id, List<CommandOption> commandOptions) {
+        this.id = id;
+        this.commandOptions = commandOptions;
+    }
+
+    @Override
+    public String getId() {
+        return id;
+    }
+
+    @Override
+    public String getLabel() {
+        for (CommandOption option : commandOptions) {
+            if (option.type() != OptionType.LABEL) continue;
+
+            return (String) option.value();
+        }
+
+        return "";
```

Comment:
Instead of returning an empty string, wouldn't it better to make it optional or something?

Reply:
i don't want to use nullable objects/things

Reply:
Optional isn't nullable. You have to explicitly check optional.isPresent if you are using Optionals

Reply:
i mean it's using nulls